src/client/datascience/raw-kernel/liveshare/hostRawNotebookProvider.ts

```diff
@@ -232,6 +270,27 @@ export class HostRawNotebookProvider
         return notebookPromise.promise;
     }
 
+    // If we need to install our dependencies now (for non-native scenarios)
+    // then install ipykernel into the interpreter or throw error
+    private async installDependenciesIntoInterpreter(
+        interpreter: PythonEnvironment,
+        ignoreDependencyCheck?: boolean,
```

Comment:
Is this set to true anywhere?

Reply:
Yes, in kernelProvider.ts (when getting the preferred kernel for VSCode notebooks, we don't display prompts or errors about dependencies missing). 
We do that only when running a cell.

Reply:
But it's a private function? Am I missing where it's called outside of this with 'ignoreDependencyCheck' set to true?

Reply:
The argument `ignoreDependencyCheck` is passed from another public function into this private function.

Reply:
The public function `getPreferredKernelForLocalConnection` has the same argument that gets passed into this private function.

Reply:
Sorry I'm still not seeing it. The function ```installDependenciesIntoInterpreter``` you added in this PR. It's only called once as far as I can tell.

Reply:
And it's called with 'false' as the value.

Reply:
In most cases its call as `false` or `undefined`. But When called from Kernel Provider in native notebooks, (which invokes `getpreferredKernelForLocalConnection`), then we pass `true`.

`false` is the default.

Reply:
Still don't see where 'installDependenciesIntoInterpreter ' is called more than once? What line is called on?

Reply:
I believe your talking about 'installDependenciesIntoInterpreter' in kernelSelector.ts. I'm talking about the copy of that function you added to hostRawNotebookProvider. It looks to me like it's only called once. And ignoreDependencies is always false.

Reply:
Yeah, I'm a bit confused on this as well. This seems like a really odd thing to be happening here in the code. Doesn't seem like a responsibility of this class.

Reply:
> I believe your talking about 'installDependenciesIntoInterpreter' in kernelSelector.ts. I'm talking about the copy of that function you added to hostRawNotebookProvider. It looks to me like it's only called once. And ignoreDependencies is always false.

Oops. Let me check the comments again.

Reply:
Sorry for the confusion, fixed

Reply:
@DonJayamanne Sorry last comment here before approving. Why are we installing the dependencies here at this point? I remember from you and I talking about this before that we wanted dependency installation at one point. But now it's happening both in the selector and here, seems confusing and not the responsibility of this class.

Reply:
I agree. However at this point there's no need to invoke the selector again as we have already selected the kernel. Happy to move this elsewhere, not sure where though?

I think we will have it at two points.
In webview & interactive window we display install instructions when selecting.
In new notebooks, we display install instructions only when starting.
As these are two separate UX  workflows I don't see how we can keep it in one place. Either way. 

Also, the code to install missting dependencies is already used in like 6 places. 
I think 1 for export, 1 for data viewer, 1 for native notebooks & the rest are just duplicates due to complex work flows.

Reply:
You mentioned invoking the selector again. I was pretty sure that your previous changes meant that when we select we should get an interpreter with dependencies installed. What's the code path where we get here with an interpreter without dependencies installed?

Reply:
Sorry I promise I'll stop fussing soon. Just seems like this is adding a very odd responsibility into this class. Does the kernelSelector code not run in the native notebook launch path now?

Reply:
Bleh I don't want to hold up overnight on this as I'm not that concerned, basically just saying that I agree with your comments way back here: https://github.com/microsoft/vscode-python/pull/11341#discussion_r413392156 when you and I were debating the same issue.

Reply:
Thats ok happy to discuss.
The the past, we didn't explictly call this method in native notebooks.
In native notebooks, we call the selector & pass a flag to ignore checking dependencies.
When starting a notebook (in native notebooks), we used to call the selector once again (that's what i'm trying to remove) & at that point the flag is false (i.e. it WILL check dependnecies).
Thus in old code, when starting a native notebook, the dependencies were checked as part of the selector.

Now that we dont have the selector anymore, we still need to ensure the dependencies are still checked.

Hope that makes sense.